### PR TITLE
Most close ephemeral peers or leak peer objects

### PR DESCRIPTION
**Pull request by Raynos**
We were leaking peer objects in prod; peer objects will
report stats in a loop. Because we 10s of thousands of
these objects we were reporting the same stat 10s of
thousands of time and saw a huge spike in about 5k
UDP writes per second.

r: @rf @jcorbin @kriskowal

### CODE REVIEW

File: node/connection.js [hunk 1]
```diff
@@ -105,6 +105,10 @@ TChannelConnection.prototype.setupSocket = function setupSocket() {
         }));
 
         if (self.ephemeral) {
+            var peer = self.channel.peers.get(self.socketRemoteAddr);
+            if (peer) {
+                peer.close(noop);
+            }
```
Comment (kriskowal) [inline node/connection.js]: This is great and fits our immediate need, but you might be able to knock a non-ephemeral peer out by issuing an ephemeral request from it.

File: node/test/lib/hyperbahn-cluster.js [hunk 2]
```diff
@@ -135,21 +136,24 @@ function HyperbahnApp(opts) {
 
     var self = this;
 
-    self.relayChannel = opts.relayChannel;
-    self.egressNodes = opts.egressNodes;
-    self.hostPort = self.relayChannel.hostPort;
+    self.__relayChannel = opts.relayChannel;
+    self.__egressNodes = opts.egressNodes;
```
Comment (kriskowal) [inline node/test/lib/hyperbahn-cluster.js]: What’s the motivation for this style change? Perhaps dunderbar is too aggressive; would one suffice?

**Comment by kriskowal** [comment]
:ok:

**Comment by rf** [comment]
lgtm

File: node/test/lib/hyperbahn-cluster.js (hunk 2 above)
Comment (Raynos) [inline node/test/lib/hyperbahn-cluster.js]: Yes; it's weird; will fix.

**Pull request merged**